llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp

```diff
@@ -3794,5 +3794,51 @@ Instruction *InstCombinerImpl::visitSelectInst(SelectInst &SI) {
   if (Instruction *I = foldBitCeil(SI, Builder))
     return I;
 
+  // Fold:
+  // (select A && B, T, F) -> (select A, (select B, T, F), F)
+  // (select A || B, T, F) -> (select A, T, (select B, T, F))
+  // if (select B, T, F) is foldable.
+  // TODO: preserve FMF flags
+  auto FoldSelectWithAndOrCond = [&](bool IsAnd, Value *A,
+                                     Value *B) -> Instruction * {
+    if (Value *V = simplifySelectInst(B, TrueVal, FalseVal,
+                                      SQ.getWithInstruction(&SI)))
+      return SelectInst::Create(A, IsAnd ? V : TrueVal, IsAnd ? FalseVal : V);
+
+    // Is (select B, T, F) a SPF?
+    if (CondVal->hasOneUse() && SelType->isIntOrIntVectorTy()) {
+      Value *LHS, *RHS;
```

Comment:
These variables are unused. It breaks the buildbot `sanitizer-x86_64-linux`.
https://lab.llvm.org/buildbot/#/builders/37/builds/29346
I will fix it later.

Reply:
Oh, it has been fixed by https://github.com/llvm/llvm-project/commit/bf312263bfee5d70a37a2269d62f08cf264ca415.
Thank you @DamonFool !